torch/lib/THC/THCCachingAllocator.cpp

```diff
@@ -159,15 +178,13 @@ struct THCCachingAllocator
 
     Block* block = it->second;
     allocated_blocks.erase(it);
-
-    bool small = block->size <= kSmallAlloc;
-    auto& free_blocks = small ? large_blocks : small_blocks;
-    try_merge_blocks(block, block->prev, free_blocks);
-    try_merge_blocks(block, block->next, free_blocks);
-
     block->allocated = false;
-    free_blocks.insert(block);
 
+    if (!block->stream_uses.empty()) {
+      return insert_events(block);
+    }
+
+    free_block(block);
```

Comment:
The `event_count` is always zero as this point. Events are only recorded in free(). There are two cases:

1) If recordStream() was called on this block it exits early in 184. `insert_events()` will increase the event_count.
2) Otherwise, recordStream() was not called, `so block->stream_uses()` is empty and no events will be recorded